### PR TITLE
Add conversion webhook

### PR DESCRIPTION
**Pull request by ldpliu**
https://github.com/open-cluster-management-io/OCM/issues/67

### CODE REVIEW

File: pkg/helpers/helpers.go [hunk 1]
```diff
@@ -335,6 +335,48 @@ func ApplyDeployment(
...
+		return fmt.Errorf("%q: %v", crdFile, err)
+	}
+	crd.(*apiextensionsv1.CustomResourceDefinition).Spec.Conversion = crdPatch.(*apiextensionsv1.CustomResourceDefinition).Spec.Conversion
+
+	if caBundle != "placeholder" {
+		//Inject ca bundle from ca-bundle-configmap
+		crd.(*apiextensionsv1.CustomResourceDefinition).Spec.Conversion.Webhook.ClientConfig.CABundle = []byte(caBundle)
+	}
+
+	_, _, err = resourceapply.ApplyCustomResourceDefinitionV1(
```
Comment (qiujian16) [inline pkg/helpers/helpers.go]: why don't we use patch directly?

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go [hunk 2]
```diff
@@ -493,6 +496,27 @@ func applyHubResources(
 		}
 	}
 
+	err = helpers.ApplyPatchedConversionCRD(
```
Comment (qiujian16) [inline pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go]: can we patch the manifest before apply?

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go [hunk 3]
```diff
@@ -493,6 +496,26 @@ func applyHubResources(
...
```
Comment (qiujian16) [inline pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go]: nit, let's add a comment here. I am still not sure whether it is a good approach. I think we could use a script to patch the manifest during make? The patch in controller make me feel it is still too complicated..

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go (hunk 2 above)
Comment (ldpliu) [reply pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go]: done

File: pkg/helpers/helpers.go (hunk 1 above)
Comment (ldpliu) [reply pkg/helpers/helpers.go]: done

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go (hunk 3 above)
Comment (ldpliu) [reply pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go]: changed to use script to patch the crd

File: hack/patch/0000_00_clusters.open-cluster-management.io_managedclustersets.crd.yaml.yaml-patch [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#This is yaml-patch config file. It's used to patch conversion webhook config to clusterset crd
+- op: add
+  path: /spec/conversion
+  value:
+    strategy: Webhook
+    webhook:
+      clientConfig:
+        service:
+          namespace: {{ .ClusterManagerNamespace }}
+          name: cluster-manager-registration-webhook
```
Comment (qiujian16) [inline hack/patch/0000_00_clusters.open-cluster-management.io_managedclustersets.crd.yaml.yaml-patch:10]: will it break cluster manager hosted mode? @xuezhaojun

**Review by qiujian16 — COMMENTED** [review]
thanks, it looks neat. Only one comments that need @xuezhaojun to take a look.

**Pull request merged**